src/exp.jl

```diff
@@ -1,121 +1,119 @@
+                        0.0013400431f0, 0.00015478022f0)
 end
-
-@inline function exp10_kernel(x::FloatType32)
-    c6 = 0.2064004987f0
-    c5 = 0.5417877436f0
-    c4 = 0.1171286821f1
-    c3 = 0.2034656048f1
-    c2 = 0.2650948763f1
-    c1 = 0.2302585125f1
-    return @horner x c1 c2 c3 c4 c5 c6
+@inline function expb_kernel(::Val{ℯ}, x::Union{Float32, SVec{W, Float32} where W})
+    return @horner(x, 1.0f0, 1.0f0, 0.5f0,
+                        0.16666415f0, 0.041666083f0,
+                        0.008375129f0, 0.0013956056f0)
+end
+@inline function expb_kernel(::Val{10}, x::Union{Float32, SVec{W, Float32} where W})
+    return @horner(x, 1.0f0, 2.3025851f0, 2.6509492f0,
+                        2.034648f0, 1.1712388f0, 
+                        0.54208815f0, 0.20799689f0)
 end
 
-"""
-    exp10(x)
-
-Compute the base-`10` exponential of `x`, that is `10ˣ`.
-"""
-@inline function exp10(d::V) where {V <: FloatType}
-    T = eltype(d)
-    q = round(T(MLOG10_2) * d)
-    qi = unsafe_trunc(fpinttype(T), q)
-
-    s = muladd(q, -L10U(T), d)
-    s = muladd(q, -L10L(T), s)
-
-    u = exp10_kernel(s)
-    u = V(dnormalize(dadd(T(1.0), dmul(u,s))))
-
-    u = ldexp2k(u, qi)
+const J_TABLE= Float64[2.0^(big(j-1)/256) for j in 1:256]
 
-    u = vifelse(d > max_exp10(T), T(Inf), u)
-    u = vifelse(d < min_exp10(T), T(0.0), u)
+@noinline function myexp_fallback(::Val{2}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp2(x[i])) end)
+end
+@noinline function myexp_fallback(::Val{ℯ}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp(x[i])) end)
+end
+@noinline function myexp_fallback(::Val{10}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp10(x[i])) end)
+end
 
-    return u
+for (func, base) in (:myexp2=>Val(2), :myexp=>Val(ℯ), :myexp10=>Val(10))
+    @eval begin
+        @inline function ($func)(x::SVec{W,T}) where {W, T<:Float64}
+            any(!(abs(x) < MAX_EXP($base, T))) && return myexp_fallback($base, x)
+            N_float = vmuladd(x, LogBo256INV($base, T), MAGIC_ROUND_CONST(T))
+            N = reinterpret(SVec{W,Int64}, N_float) % Int32
+            N_float = vsub(N_float, MAGIC_ROUND_CONST(T))
+            
+            r = vmuladd(N_float, LogBo256U($base, T), x)
+            r = vmuladd(N_float, LogBo256L($base, T), r)
+            js = vload(stridedpointer(J_TABLE), (N&255,))
+            k = N >> 8
+            
+            small_part = reinterpret(SVec{W, Int64}, vmuladd(js, expm1b_kernel($base, r), js))
+            twopk = rem(vadd(k,53), UInt64) << 52
+            return reinterpret(SVec{W, T}, vadd(twopk, small_part))
+        end
+        
+        @inline function ($func)(x::SVec{W,T}) where {W, T<:Float32}
+            any(!(abs(x) < MAX_EXP($base, T))) && return myexp_fallback(x)
+            N_float = vmuladd(x, LogBINV($base, T), MAGIC_ROUND_CONST(T))
+            N = reinterpret(SVec{W,Int32}, N_float)
```

Comment:
`reinterpret(T1, v::SVec{W,T2})` should work.

You should be able to make the function signatures just `x::Union{T,SVec{<:Any,T}}`

Reply:
The reason we want scalar definitions is that the test suite isn't vectorized.

Reply:
An alternative hack would be
```julia
exp(x) = exp((Core.VecElement(x),))[1]
```
But I don't believe Julia's autovectorizer would vectorize that, so allowing for that possibility is why we prefer to actually define the scalar version using scalars instead of length-1 vectors.